### ISSUE TITLE
Opening Prometheus query editor throws console error

### ISSUE THREAD

**Issue by baaym** [grafana/grafana#13192]
Hi all,

When I add a new panel based on a Prometheus datasource, the query editor doesn't appear. The console throws the following error in Chrome:
```
angular.js:14700 TypeError: Cannot use 'in' operator to search for 'opacity' in undefined
    at new o (index.js:14598)
    at new g (index.js:15538)
    at Object.t.edit (index.js:19037)
    at Object.r [as link] (code_editor.ts:85)
    at angular.js:1385
    at angular.js:10539
    at Ae (angular.js:10545)
    at h (angular.js:9934)
    at a (angular.js:9174)
    at a (angular.js:9177) "<code-editor content="ctrl.target.expr" datasource="ctrl.datasource" on-change="ctrl.refreshMetricData()" get-completer="ctrl.getCompleter()" data-mode="prometheus" code-editor-focus="ctrl.isLastQuery" class=" ace_editor ace-tm">"
```

## Grafana installation info
- Grafana version 5.2.3 running on Debian
- Browsers tested:
  - Chrome 69.0.3497.81
  - Firefox 62.0

## Screenshot:
![image](https://user-images.githubusercontent.com/705382/45224099-6cde4500-b2b9-11e8-86e4-782c57399c6a.png)

**Comment by marefr** [comment]
@baaym not able to reproduce on v5.2.3 [here](https://play.grafana.org/d/000000063/prometheus-templating?orgId=1). Maybe it's because of you're using a certain type of query? Please include your full query (replace any sensitive information). You should be able to get a hold of the query using panel menu -> More -> Panel Json - please include the full json content of the "targets" property. Also, what version of Prometheus are you using? Thanks

**Comment by baaym** [comment]
Hi @marefr,

Thanks for having a look at this ticket.

I don't think it could be related to the query, as this happens on creation of a fresh panel.

## More detailed reproduction steps
- Create new 'graph' panel on dashboard
- Change to a Prometheus datasource (Prometheus is not default)
- Error shows in console

## Prometheus version
Version 2.3.1
Revision 188ca45bd85ce843071e768d855722a9d9dabe03

## Panel Json targets property
`"targets": [
    {
      "policy": "default",
      "resultFormat": "time_series",
      "orderByTime": "ASC",
      "tags": [],
      "groupBy": [
        {
          "type": "time",
          "params": [
            "$__interval"
          ]
        },
        {
          "type": "fill",
          "params": [
            "null"
          ]
        }
      ],
      "select": [
        [
          {
            "type": "field",
            "params": [
              "value"
            ]
          },
          {
            "type": "mean",
            "params": []
          }
        ]
      ],
      "refId": "A",
      "expr": "",
      "intervalFactor": 1,
      "format": "time_series"
    }
  ],`

**Comment by marefr** [comment]
Unable to reproduce on play.grafana.org (v5.2.3) or locally (v5.3-pre1 7a117a6) using your more detailed steps. 

Could it be due to you have a refresh of each 10 seconds? 
Do you use some special character in your datasource name?
Maybe possible for you to include more detailed stack trace when the problem happens?
If you have chrome dev tools open and do a empty cache and hard reload (see screenshot below) - does that make any difference?

![image](https://user-images.githubusercontent.com/1668778/45283183-76e08d80-b4dd-11e8-982c-f4472cec20af.png)

**Comment by marefr** [comment]
@baaym  Update: your json targets property looks to me you're using an influxdb datasource. Have you by any chance renamed a prometheus datasource to a name that was earlier used by an influxdb datasource?

**Comment by baaym** [comment]
Hi @marefr,

If the targets property looks like an influxdb datasource, that might be due to the fact that my system's default datasource is influxdb.

Could it be that the following is the case?
- New panel is created. Since default datasource is influxdb, an empty influxdb query is created and written in the 'targets' property
- Panel datasource is changed to Prometheus. It tries to do something with the existing query which will fail.

However, this error also occurs when I delete all queries in the panel before switching the datasource.

Another fun fact: When I was copying the panel JSON I noticed there was no input field I could copy/paste from. The 'Copy to Clipboard' button works fine, which is why I could paste it here.
However I now see that opening the Panel JSON window throws the exact same error as switching to the Prometheus datasource.

`TypeError: "right-hand side of 'in' should be an object, got undefined"
 o    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:43:441801
 g    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:43:464594
 edit https://redacted/public/build/vendor.6aa12092d3333cb24394.js:43:531824
 r    https://redacted/public/build/app.6aa12092d3333cb24394.js:8:386575
 K    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:6121
 xe   https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:41976
 Ae   https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:42035
 h    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:34379
 a    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29904
 a    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29921
 a    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29921
 h    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:34323
 a    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29904
 a    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29921
 U    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:29421
 l    https://redacted/public/build/app.6aa12092d3333cb24394.js:8:854
 c    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:83932
 i    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:21868
 n    https://redacted/public/build/vendor.6aa12092d3333cb24394.js:87:23332 <code-editor content="json" data-mode="json" data-max-lines="20" class=" ace_editor ace-tm"> angular.js:14700 `

Using dev tools in Firefox all cached were disabled and cleared.

Regarding the name of the datasource: It's 'gitlab-review-apps'. I also had a go at renaming it so it doesn't have any hyphens - but that didn't result in anything.

**Comment by baaym** [comment]
Update:

In Chrome the sourcemap is resolved and the error translates to this:

`angular.js:14700 TypeError: Cannot use 'in' operator to search for 'opacity' in undefined
    at new o (index.js:14598) // node_modules/brace/index.js
    at new g (index.js:15538)
    at Object.t.edit (index.js:19037)
    at Object.r [as link] (code_editor.ts:85)
    at angular.js:1385
    at angular.js:10539
    at Ae (angular.js:10545)
    at h (angular.js:9934)
    at a (angular.js:9174)
    at a (angular.js:9177) "<code-editor content="json" data-mode="json" data-max-lines="20" class=" ace_editor ace-tm">"`

Line 14597 and 14598:

`    if (isIE8 === undefined)
        isIE8 = !("opacity" in this.element.style);`

**Comment by marefr** [comment]
Just tried the following but not able to reproduce using Chrome 69.0.3497.81:
* New panel is created. Since default datasource is influxdb, an empty influxdb query is created and written in the 'targets' property
* Panel datasource is changed to Prometheus. It tries to do something with the existing query which will fail. 

Could it be related to any extension you've enabled? Make any difference using Incognito mode?

**Comment by marefr** [comment]
@torkelo any additional thoughts/inputs on this?

**Comment by baaym** [comment]
In Firefox and Chrome incognito mode the same error occurs. I also checked Edge to be sure and there it happens as well.

I actually think it's related to the `code-editor` component. Existing influxdb widgets that work fine also throw an exception when I want to load their panel JSON.

**Comment by baaym** [comment]
I tried on play.grafana.org but I cannot reproduce either. However I see that it's running `2.5.4` already. Let me try upgrading the installation and see if that resolves it.

**Comment by marefr** [comment]
Not able to reproduce on Windows using Edge.

Do you have any plugins installed? If so, which ones?

**Comment by baaym** [comment]
There seems to be something happening on the firewall between the server and my desktop. I updated our installation to 2.5.4 to match 'play.grafana.org', and I compared both `vendor.4f5454f867a0cc2fe8dd.js` files. It appears that when downloading scripts on my installation, all `http` references have been replaced with `https`:

![image](https://user-images.githubusercontent.com/705382/45290474-f7a88500-b4ef-11e8-922a-c637110ab5df.png)

While on the server, they're as they should be:
![image](https://user-images.githubusercontent.com/705382/45290690-8e754180-b4f0-11e8-9ea0-3afd7d912bad.png)

I will have to chase someone here internally because this smells like a broken configuration somewhere. In the meanwhile this issue can be closed, because obviously if something messes with the scripts then that's not your problem ;)
